textile-utils-api-v1/src/main/java/net/textilemc/textile/utils/util/Identifier.java

```diff
@@ -0,0 +1,34 @@
+package net.textilemc.textile.utils.util;
+
+import net.minecraft.nbt.CompoundTag;
+
+public class Identifier {
+	private final String namespace;
+	private final String path;
+
+
+	public Identifier (String namespace, String path) {
+		this.namespace = namespace;
+		this.path = path;
+	}
+
+	public String getNamespace() {
+		return namespace;
+	}
+
+	public String getPath() {
+		return path;
+	}
+
+	public CompoundTag toTag(CompoundTag tag) {
```

Comment:
You don’t need a CompoundTag for this. That just wastes bytes 
Store it as a StringTag

Reply:
What’s the use case of CompoundTag vs Tag?

Reply:
CompoundTag is a maplike tag
Tag is the base class for tags